packages/flutter/test/material/dropdown_menu_test.dart

```diff
@@ -882,6 +918,94 @@ void main() {
+      'when it is tapped', (WidgetTester tester) async {
+    final ThemeData themeData = ThemeData();
+
+    Widget buildDropdownMenu({bool? requestFocusOnTap}) => MaterialApp(
+      theme: themeData,
+      home: Scaffold(
+        body: Column(
+          children: <Widget>[
+            DropdownMenu<TestMenu>(
+              requestFocusOnTap: requestFocusOnTap,
+              dropdownMenuEntries: menuChildren,
+            ),
+          ],
+        ),
+      ),
+    );
+
+    // Test default condition.
+    await tester.pumpWidget(buildDropdownMenu());
+    await tester.pump();
+
+    final Finder textFieldFinder = find.byType(TextField);
+    final TextField result = tester.firstWidget<TextField>(textFieldFinder);
+    expect(result.canRequestFocus, false);
+  }, variant: TargetPlatformVariant.mobile());
+
+  testWidgets('The text input field should be focused on desktop platforms '
+      'when it is tapped', (WidgetTester tester) async {
+    final ThemeData themeData = ThemeData();
+
+    Widget buildDropdownMenu({bool? requestFocusOnTap}) => MaterialApp(
+      theme: themeData,
+      home: Scaffold(
+        body: Column(
+          children: <Widget>[
+            DropdownMenu<TestMenu>(
+              requestFocusOnTap: requestFocusOnTap,
+              dropdownMenuEntries: menuChildren,
+            ),
+          ],
+        ),
+      ),
+    );
+
+    await tester.pumpWidget(buildDropdownMenu());
+    await tester.pump();
+
+    final Finder textFieldFinder = find.byType(TextField);
+    final TextField result = tester.firstWidget<TextField>(textFieldFinder);
+    expect(result.canRequestFocus, true);
+  }, variant: TargetPlatformVariant.desktop());
+
+  testWidgets('If the requestFocusOnTap is true, the text input field can request focus, '
+    'otherwise it cannot request focus', (WidgetTester tester) async {
+    final ThemeData themeData = ThemeData();
+
+    Widget buildDropdownMenu({required bool requestFocusOnTap}) => MaterialApp(
+      theme: themeData,
+      home: Scaffold(
+        body: Column(
+          children: <Widget>[
+            DropdownMenu<TestMenu>(
+              requestFocusOnTap: requestFocusOnTap,
+              dropdownMenuEntries: menuChildren,
+            ),
+          ],
+        ),
+      ),
+    );
+
+    // Set requestFocusOnTap to true.
+    await tester.pumpWidget(buildDropdownMenu(requestFocusOnTap: true));
+    await tester.pump();
+
+    final Finder textFieldFinder = find.byType(TextField);
+    final TextField result = tester.firstWidget<TextField>(textFieldFinder);
+    expect(result.canRequestFocus, true);
+
+    // Set requestFocusOnTap to false.
+    await tester.pumpWidget(buildDropdownMenu(requestFocusOnTap: false));
+    await tester.pump();
+
+    final Finder textFieldFinder1 = find.byType(TextField);
+    final TextField result1 = tester.firstWidget<TextField>(textFieldFinder1);
+    expect(result1.canRequestFocus, false);
+  }, variant: TargetPlatformVariant.all());
```

Comment:
Presumably `variant: TargetPlatformVariant.all()` is the default?

Reply:
Just want to test that the dropdown menu works on all platforms when a value of `requestFocusOnTap` is provided.